File: app/src/main/java/com/mrswimmer/shift/di/module/SharedPreferencesModule.java
package com.mrswimmer.shift.di.module;

import android.content.Context;
import android.content.SharedPreferences;

import com.mrswimmer.shift.domain.interactor.SettingsService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SharedPreferencesModule {

    private Context context;

    public SharedPreferencesModule(Context context) {
        this.context = context;
    }

    @Provides
    public Context context() {
        return context;
    }

    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences() {
        return context.getSharedPreferences("settings", Context.MODE_PRIVATE);
    }

    @Provides
    @Singleton
    SettingsService settingsService(SharedPreferences sharedPreferences) {
        return new SettingsService(sharedPreferences);
    }
}

File: app/src/main/java/com/mrswimmer/shift/di/module/SharedPreferencesModule.kt
package com.bignerdranch.android.osm.di.module

import android.content.Context
import android.content.SharedPreferences
import com.mrswimmer.shift.domain.interactor.SettingsService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SharedPreferencesModule(private var context: Context) {

    @Provides
    fun context(): Context {
        return context
    }

    @Provides
    @Singleton
    fun provideSharedPreferences(): SharedPreferences {
        return context.getSharedPreferences("settings", Context.MODE_PRIVATE)
    }

    @Provides
    @Singleton
    fun settingsService(sharedPreferences: SharedPreferences): SettingsService {
        return SettingsService(sharedPreferences)
    }

}